Allow Notifier component to limit amount of notifications to return.

akkaweb opened the pull request:
Changes to the method `getNotifications()` to accept a 3rd paremeter `$limit = null` and make the necessary code changes to accept limit. README.md also changed to reflect Component changes.

akkaweb committed:
Updating README.md to reflect new changes to the Component

README has instructions for method `allNotificationList()`, but component does not have this method. Replacing instruction with `getNotifications()` instead.

akkaweb committed:
Allow notifications limit to be passed to getNotifications()

Change `getNotifications()` method to accept a 3rd parementer for $limit and make code changes to pull from DB

bobmulder commented:
Hi @akkaweb,

First of all, thank you for the PR! However, I have some suggestions for you. 

1) As you see Travis failed. The reason you can see why: [here](https://travis-ci.org/cakemanager/cakephp-notifier/jobs/118056441). You didn't handle any code-standard. Thats why the [file list](https://github.com/cakemanager/cakephp-notifier/pull/8/files) is f*cked up, so I can't see that clear what code you added/changed. If you're not familiar with code standards let me know.

2) No tests are added for this functionality. Thats why coveralls came in to say that the coverage decreased. Please add tests to see if the method works without limit, and with multiple limits.

If you have questions, feel free to chat me!

Looking forward to your changes! Good luck!

akkaweb commented:
My apologies about that. I mostly develop for Drupal and my IDE is setup to work with their coding standards. I will fix and resubmit.

akkaweb committed:
Update NotifierComponent.php

Comply with CakePHP coding standards

akkaweb committed:
Update NotifierComponentTest.php

Added 2 test cases for the newly added $limit parameter

akkaweb commented:
I updated the code to comply with CakePHP standards. I will watch Travis to see if it allows it to pass now. Test cases have been added now.

bobmulder commented on README.md:
```diff
@@ -96,11 +96,20 @@ Of course you want to get a list of notifications per user. Here are some exampl
     
-    // getting a list of all unread notifications
-    $this->Notifier->allNotificationList(2, true);
+    // getting a list of all unread notifications for user with ID 2
```
I documented with 'id', here you did it with 'ID' and below you typed 'Id'. What will it be? I suggest 'id'.

bobmulder commented on src/Controller/Component/NotifierComponent.php:
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Why these empty lines?

bobmulder commented:
@akkaweb Thank you! Good work! I added some comments you should look at. Also the code-coverage gave one error: https://travis-ci.org/cakemanager/cakephp-notifier/jobs/118129153#L278. If you can fix these, it will be okay!


As I see, you added a tirth parameter the the method, which is good enough. But I was tinking, to be prepared of the future, we could make the tirth parameter an array, with the option `limit`. What do you think about that?

akkaweb committed:
Update NotifierComponent.php

Change 3rd parameter to an array which allows the possibility for future expansion. Currently it only supports 'limit' - $options['limit']

akkaweb committed:
Update README.md

Making changes to documentation to include $options array as 3rd parameter. Currently it only supports 'limit' - $options['limit']

akkaweb committed:
Update NotifierComponentTest.php

Updating test cases to accept array as 3rd parameter with 'limit' as only key support at the moment

akkaweb commented:
I totally agree about the 3rd parameter being an array. Thanks for catching that! I put that change into place, changed test cases to reflect that update and I also updated code-coverage to include the 3rd option. Hopefully all the tests will now pass. If any issues or if you would something changed let me know. Thanks

akkaweb committed:
Update NotifierComponent.php

Travis check failed to whitespace at the end of line and a non-existent space after If. Should pass now!

akkaweb committed:
Update NotifierComponent.php

bobmulder commented:
Sorry for leaving this behind @akkaweb. Lets finish this :)

I see you did 9 commits. 

I would like to see the commits squased (read more at http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html and http://couscous.io/contributing.html)

Greetz,

Bob

akkaweb commented:
@bobmulder Sure! I will give it a go soon! Thanks!

dereuromark commented on README.md:
```diff
@@ -90,17 +90,29 @@ be defined here.
 Of course you want to get a list of notifications per user. Here are some examples:
 
-    // getting a list of all notifications of the current logged in user
+    // getting a list of all notifications of the current Logged In user
```
I dont think that change was valid :)

dereuromark commented on src/Controller/Component/NotifierComponent.php:
````diff
@@ -88,12 +89,17 @@ public function setController($controller)
      * ```
      * @param int|null $userId Id of the user.
      * @param bool|null $state The state of notifications: `true` for unread, `false` for read, `null` for all.
+     * @param array|empty $options Currently supports 'limit', which limits returned notifications
````
`|empty` should be removed. its invalid doc block type markup.
